### PR TITLE
Added method_Piyavskogo

### PR DESCRIPTION
**Pull request by SokolovaNastya**

### CODE REVIEW

**Comment by akharche** [comment]
Please, add at least 1 reviewer for PR

File: Src/Optimization/Function.cpp [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	b = _b;
+	c = _c;
+	d = _d;
+}
+double Function::operator()(double x)
+{
+	return calculate(x);
+}
+
+double Function::calculate(double x)
```
Comment (Zebraside) [inline Src/Optimization/Function.cpp]: IMO, name Function with hardcoded calculations is quite confusing. Please use Function class as interface or rename this class to smth more meaningful.

**Comment by AlexanderSergeev** [comment]
Code review:
1. Is there a way we can call function parameters in a better way so that they can be recognized? (instead of using a, b, c, d names which do not have any meaning). 
This way it will be easier to understand what the function actually does.

2. It seems dangerous to use a 'while(true)' cycle in that place. Can it be substituted or maybe used inside the actual function so that we could see the return statements and know that we wouldn't have the infinite loop problem?

**Review by AlexanderSergeev — COMMENTED** [review]
1. Is there a way we can call function parameters in a better way so that they can be recognized? (instead of using a, b, c, d names which do not have any meaning).
This way it will be easier to understand what the function actually does.

2. It seems dangerous to use a 'while(true)' cycle in that place. Can it be substituted or maybe used inside the actual function so that we could see the return statements and know that we wouldn't have the infinite loop problem?

**Review by AlexanderSergeev — APPROVED** [review]
Changes approved.

File: Src/Optimization/Function.cpp (hunk 1 above)
Comment (pavlanast) [reply Src/Optimization/Function.cpp]: Fixed

**Pull request merged**